spritzle/torrent.py

```diff
@@ -0,0 +1,73 @@
+#
+# spritzle/torrent.py
+#
+# Copyright (C) 2016 Andrew Resch <andrewresch@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.    If not, write to:
+#   The Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor
+#   Boston, MA    02110-1301, USA.
+#
+
+import asyncio
+import logging
+
+import libtorrent as lt
+
+log = logging.getLogger('spritzle')
+
+
+class AlertException(Exception):
+    def __init__(self, alert):
+        self.alert = alert
+
+
+class Torrent(object):
```

Comment:
This needs a test.

Reply:
I added a simple test, not sure about the `asyncio.sleep` calls in it, but not sure a better way to context switch so the task can run. I feel better about the tests in `resource/test_torrent.py` since they run the code a lot closer to how it runs in production (though end up testing much more than this class.)